client.go

```diff
@@ -133,8 +133,11 @@ func (c *Client) startPrometheus() {
 		return
 	}
 
+	// Create a new mux to avoid an error serving the same path in multiple instances
+	mux := http.NewServeMux()
+
 	c.log.Warnf("Exposing Prometheus metrics on port %d", c.opts.statsPort)
-	http.Handle("/metrics", promhttp.Handler())
+	mux.Handle("/metrics", promhttp.Handler())
```

Comment:
Hmm, we do only really need one of these, not one for each client since the prom stuff are all shared between all running instnaces.

I didnt intend multiple clients in the same process but makes sense - so maybe we can somehow start this metric only once, or do you also give each a unique port?

Reply:
Each port would serve the same statistics, it wouldnt be correct at all - so I think the better fix is to support disabling prometheus on clients so you can stop it in tests?

Reply:
Yes, removing Prom for testing does the job. Maybe it's a documentation issue.

Should I close this now?

Reply:
if you set prom port to 0 it wont start, that should work for you - tst and let me know pls

Reply:
Yes, I confirm. I instantiate 2 clients. Setting one of them or both to 0 solves this issue

Reply:
Cool then I think we can close these